### ISSUE TITLE
observe doesn't work when using findFromPublication()

### ISSUE THREAD

**Issue by inbaldoron** [percolatestudio/find-from-publication#4]
findFromPublication works great for me when I use it from within a template but when I try using the observe function on the cursor I don't hit the added code when adding new documents to the db (on initial page load the added function does happen for all items that are already in the collection, but afterwards it doesn't).

```
// client code
Meteor.subscribe('allPosts', {
    onReady: function() {      
      Posts.findFromPublication('allPosts', {}).observe({      
        added: function(doc) {          
          // this line isn't hit when adding new documents
        }
      });          
    },
    onError: function() {
      console.log("error");      
    }
  });  
```

**Comment by tmeasday** [comment]
Ahh. That's because it does a "join" from the metadata collection into posts. If you had this in a reactive context (i.e. autorun) it wouldn't be a problem (although I suspect an `observer` might not be what you want in that case).

Basically think of it like

```
Posts.findFromPublication = function() {
  var postIds = Metadata.findIds('posts');
  return Posts.find(postIds);
}
```

Does that make sense? Sorry if I'm not explaining it very well...

**Comment by inbaldoron** [comment]
yeah that makes perfect sense, I'm trying to think about a way to solve this, since observe is quite a common use case for me (and to others I think)
I tried changing the code and making the ids var reactive in all sorts of ways but it didn't seem to have much effect.
any ideas?

**Comment by tmeasday** [comment]
Hmmm... makes me think that a joining observer might be useful on more than just the server....

I guess you basically just want an `added` callback? You could do something like:

``` js
var addedPostIds = [];
Tracker.autorun(function() {
  Posts.findFromPublication(...).forEach(function(post) {
    if (!_.include(addedPostIds, post._id) {
      added(post);
      addedPostIds.push(post);
    }
  });
});
```

**Comment by inbaldoron** [comment]
Thanks, that does the trick as far as an added callback, but it's not the most efficient and general solution.
I'll give it another try at implementing something more general
